### DISCUSSION TITLE
Dependency injection of EF DbContext

### DISCUSSION THREAD

**Discussion by mohaaron** [PrismLibrary/Prism#2341] [WPF]
I am somewhat new to WPF and Prism and have a question about how DI works with regards to injecting an EF DbContext.

I have a WPF app setup and in the App.xaml.cs I have my application services registered for DI. I've noticed that I did not have to explicitly register the context class to for it to be injectable. It seems to be automatically registered somehow.

Here is the code for the DI registrations. I expected that I would have to include the DbContext, but it works the way it is. Can anyone tell me why it works without being registered?

````
protected override void RegisterTypes(IContainerRegistry containerRegistry)
{
       //containerRegistry.Register<IpiphiniContext>(); // Why don't we need this?
        containerRegistry.RegisterSingleton<IAuthenticationService, AuthenticationService>();
        containerRegistry.RegisterSingleton<IAuthenticatedUser, AuthenticatedUser>();
        containerRegistry.RegisterSingleton<IAssetService, AssetService>();
        containerRegistry.RegisterSingleton<IWorkOrderService, WorkOrderService>();
}
```

**Comment by brianlagunas** [comment]
Because containers can resolve concrete types without registration
